gms/gossiper.cc

```diff
@@ -441,12 +448,12 @@ future<> gossiper::handle_echo_msg(const rpc::client_info& cinfo, std::optional<
         logger.warn("handle_echo_msg: advertise_myself=false");
     }
     if (!respond) {
-        return make_exception_future(std::runtime_error("Not ready to respond gossip echo message"));
+        co_await coroutine::return_exception(std::runtime_error("Not ready to respond gossip echo message"));
     }
-    return make_ready_future<>();
 }
 
 future<> gossiper::handle_shutdown_msg(const rpc::client_info& cinfo, std::optional<int64_t> generation_number_opt) {
+    co_await update_address_map(cinfo);
```

Comment:
Do we really need to update the map on each and every gossiper messages? Here for instance a node reports that it is going to shutdown. It may boot with another IP. Why is it important to update the map here? Shouldn't we update it only on a first message the gossiper sends (echo?)?

Reply:
If the peer behaves itself, it should suffice.
Updating the address_map is supposed to be lightweight and apply only when there is no such mapping.
But with using the raft_address_map we can reconsider that and update only on echo or syn (which is sent also to unreachable endpoints).